### PR TITLE
Create workflow to auto-publish to NPM

### PR DESCRIPTION
**Pull request by luacmartins**
This PR implements a GitHub workflow that automatically publishes a new version of the library to NPM every time a PR is merged to main.

Tests
1. Merge this PR
2. Verify that it triggers the publish workflow
3. Verify that a new release is created and published to NPM

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    version:
+        runs-on: ubuntu-latest
+
+        # OSBotify will update the version on `main`, so this check is important to prevent an infinite loop
+        if: ${{ github.actor != 'OSBotify' }}
+
+        steps:
+            - uses: actions/checkout@v3
+              with:
+                ref: main
```
Comment (roryabraham) [inline .github/workflows/publish.yml:22]: Gotta pass OSBotify token to `actions/checkout` or else pushing to protected branch won't work. It's an org secret so this should just work:

```suggestion
                ref: main
                # The OS_BOTIFY_COMMIT_TOKEN is a personal access token tied to osbotify, which allows him to push to protected branches
                token: ${{ secrets.OS_BOTIFY_COMMIT_TOKEN }}
```

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+            - name: Set new version in GitHub ENV
+              run: echo "NEW_VERSION=$(jq '.version' package.json)" >> $GITHUB_ENV
+
+            - name: Push branch and publish tags
+              run: git push --set-upstream origin ${{ env.BRANCH_NAME }} && git push --tags
+
+            - name: Create pull request
+              run: |
+                gh pr create \
```
Comment (roryabraham) [inline .github/workflows/publish.yml]: Let's not use automated / self-merged PRs. Too noisy. Instead, let's create an issue to update `OSBotify`'s permissions in this repo to match his permissions in E/App, react-native-onyx, react-native-live-markdown, and other repos, and then push the version bump directly to main w/o a pull request.

**Review by roryabraham — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (luacmartins) [reply .github/workflows/publish.yml:22]: Oh I have an outdated version of this workflow. I'll update it to match the one used in Onyx

File: .github/workflows/publish.yml (hunk 2 above)
Comment (luacmartins) [reply .github/workflows/publish.yml]: @justinpersaud can we update OSBotify's permission as described above? Also happy to create a separate issue if you'd like

**Comment by luacmartins** [comment]
Updated!

File: .github/workflows/publish.yml (hunk 2 above)
Comment (justinpersaud) [reply .github/workflows/publish.yml]: I'll do it now. Can you create an issue and assign me too? I have to edit the branch protections on the repo and it's going to create a GH for someone to investigate why I'm doing that.

Comment (justinpersaud) [reply .github/workflows/publish.yml]: added it in

File: .nvmrc [hunk 3]
```diff
@@ -0,0 +1 @@
+20.14.0
```
Comment (justinpersaud) [inline .nvmrc]: NAB, we updated to 20.15.1 recently everywhere
```suggestion
20.15.1
```

**Review by justinpersaud — DISMISSED** [review]

File: .github/workflows/publish.yml (hunk 2 above)
Comment (luacmartins) [reply .github/workflows/publish.yml]: Created here - https://github.com/Expensify/Expensify/issues/423724

**Comment by luacmartins** [comment]
Updated node version!

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+                gpg --import OSBotify-private-key.asc
+              env:
+                LARGE_SECRET_PASSPHRASE: ${{ secrets.LARGE_SECRET_PASSPHRASE }}
+
+            - name: Set up git for OSBotify
+              run: |
+                git config --global user.signingkey 367811D53E34168C
+                git config --global commit.gpgsign true
+                git config --global user.name OSBotify
+                git config --global user.email infra+osbotify@expensify.com
```
Comment (roryabraham) [inline .github/workflows/publish.yml:39]:
```suggestion
                git config --global user.email infra+osbotify@expensify.com

```

File: .github/workflows/publish.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            - uses: actions/setup-node@v4
+              with:
+                node-version-file: ".nvmrc"
+                registry-url: "https://registry.npmjs.org"
```
Comment (roryabraham) [inline .github/workflows/publish.yml]: Is this `registry-url` necessary? We don't use it in NewDot

**Review by roryabraham — CHANGES_REQUESTED** [review]
minor thing but it's too :eye_twitch: to be NAB

File: .github/workflows/publish.yml (hunk 5 above)
Comment (luacmartins) [reply .github/workflows/publish.yml]: I think the default is already `https://registry.npmjs.org` so I think we can remove this as it wouldn't really override anything.

**Comment by luacmartins** [comment]
Updated

**Review by roryabraham — COMMENTED** [review]
LGTM but we need to wait for the permissions update before merging

**Review by roryabraham — APPROVED** [review]

**Comment by justinpersaud** [comment]
Permissions were already updated

**Comment by roryabraham** [comment]
Nice, let's give this a go then

**Comment by luacmartins** [comment]
Hmm maybe we missed some permission @justinpersaud? I'm seeing:

```
Run git push origin main && git push --tags
remote: error: GH013: Repository rule violations found for refs/heads/main.        
remote: Review all repository rules at http://github.com/Expensify/react-native-qrcode-svg/rules?ref=refs%2Fheads%2Fmain        
remote: 
remote: - Changes must be made through a pull request.        
remote: 
To https://github.com/Expensify/react-native-qrcode-svg
 ! [remote rejected] main -> main (push declined due to repository rule violations)
error: failed to push some refs to 'https://github.com/Expensify/react-native-qrcode-svg'
Error: Process completed with exit code 1.
```

**Comment by roryabraham** [comment]
Might also check that the OS_BOTIFY_COMMIT_TOKEN is accessible in this repo

**Pull request merged**